File: files-objc/SettingsViewController.m
// SettingsViewController.m
#import "SettingsViewController.h"
#import "FileListViewController.h"
#import "AboutViewController.h"
#import "UpdateLogViewController.h"

@interface SettingsViewController ()

@property (nonatomic, strong) NSArray *settingsItems;

@end

@implementation SettingsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupUI];
    [self setupSettingsItems];
}

- (void)setupUI {
    self.title = @"设置";
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"SettingsCell"];
    
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:self action:@selector(dismissSettings)];
}

- (void)setupSettingsItems {
    self.settingsItems = @[
        @"显示隐藏文件",
        @"文件排序方式", 
        @"关于",
        @"更新日志"
    ];
}

- (void)dismissSettings {
    [self dismissViewControllerAnimated:YES completion:nil];
}

#pragma mark - UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.settingsItems.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"SettingsCell" forIndexPath:indexPath];
    cell.textLabel.text = self.settingsItems[indexPath.row];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    
    // 显示隐藏文件开关
    if (indexPath.row == 0) {
        UISwitch *switchView = [[UISwitch alloc] init];
        switchView.on = self.fileListVC.showHiddenFiles;
        [switchView addTarget:self action:@selector(toggleHiddenFiles:) forControlEvents:UIControlEventValueChanged];
        cell.accessoryView = switchView;
        cell.selectionStyle = UITableViewCellSelectionStyleNone;
    } else {
        cell.accessoryView = nil;
        cell.selectionStyle = UITableViewCellSelectionStyleDefault;
    }
    
    return cell;
}

#pragma mark - UITableViewDelegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    switch (indexPath.row) {
        case 1: // 文件排序方式
            [self showSortOptions];
            break;
        case 2: // 关于
            [self showAboutPage];
            break;
        case 3: // 更新日志
            [self showUpdateLog];
            break;
        default:
            break;
    }
}

- (void)toggleHiddenFiles:(UISwitch *)sender {
    self.fileListVC.showHiddenFiles = sender.on;
    [self.fileListVC loadDirectoryContents];
}

- (void)showSortOptions {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"排序方式" message:nil preferredStyle:UIAlertControllerStyleActionSheet];
    
    [alert addAction:[UIAlertAction actionWithTitle:@"按名称" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self showSuccessAlert:@"已设置为按名称排序"];
    }]];
    
    [alert addAction:[UIAlertAction actionWithTitle:@"按日期" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self showSuccessAlert:@"已设置为按日期排序"];
    }]];
    
    [alert addAction:[UIAlertAction actionWithTitle:@"按大小" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [self showSuccessAlert:@"已设置为按大小排序"];
    }]];
    
    [alert addAction:[UIAlertAction actionWithTitle:@"取消" style:UIAlertActionStyleCancel handler:nil]];
    
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)showAboutPage {
    AboutViewController *aboutVC = [[AboutViewController alloc] init];
    [self.navigationController pushViewController:aboutVC animated:YES];
}

- (void)showUpdateLog {
    UpdateLogViewController *updateLogVC = [[UpdateLogViewController alloc] init];
    [self.navigationController pushViewController:updateLogVC animated:YES];
}

- (void)showSuccessAlert:(NSString *)message {
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"成功" message:message preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}

@end

File: files-swift/SettingsViewController.swift
import UIKit

class SettingsViewController: UITableViewController {
    
    weak var fileListVC: FileListViewController?
    
    let settingsItems = [
        "关于",
        "显示隐藏文件",
        "文件排序方式",
        "更新日志"
    ]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
    }
    
    private func setupUI() {
        title = "设置"
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "SettingsCell")
        
        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(dismissSettings))
    }
    
    @objc private func dismissSettings() {
        dismiss(animated: true)
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return settingsItems.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "SettingsCell", for: indexPath)
        cell.textLabel?.text = settingsItems[indexPath.row]
        cell.accessoryType = .disclosureIndicator
        
        // 显示隐藏文件开关
        if indexPath.row == 1 {
            let switchView = UISwitch()
            switchView.isOn = fileListVC?.showHiddenFiles ?? false
            switchView.addTarget(self, action: #selector(toggleHiddenFiles(_:)), for: .valueChanged)
            cell.accessoryView = switchView
            cell.selectionStyle = .none
        } else {
            cell.accessoryView = nil
            cell.selectionStyle = .default
        }
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        
        switch indexPath.row {
        case 0:
            showAboutPage()
        case 2:
            showSortOptions()
        case 3:
            showUpdateLog()
        default:
            break
        }
    }
    
    @objc private func toggleHiddenFiles(_ sender: UISwitch) {
        fileListVC?.showHiddenFiles = sender.isOn
        fileListVC?.loadDirectoryContents()
    }
    
    private func showAboutPage() {
        let aboutVC = AboutViewController()
        navigationController?.pushViewController(aboutVC, animated: true)
    }
    
    private func showSortOptions() {
        let alert = UIAlertController(title: "排序方式", message: nil, preferredStyle: .actionSheet)
        
        alert.addAction(UIAlertAction(title: "按名称", style: .default))
        alert.addAction(UIAlertAction(title: "按日期", style: .default))
        alert.addAction(UIAlertAction(title: "按大小", style: .default))
        alert.addAction(UIAlertAction(title: "取消", style: .cancel))
        
        present(alert, animated: true)
    }
    
    private func showUpdateLog() {
        let updateLogVC = UpdateLogViewController()
        navigationController?.pushViewController(updateLogVC, animated: true)
    }
}